### DISCUSSION TITLE
Search users  by roles API

### DISCUSSION THREAD

**Discussion by ashep** [zitadel/zitadel#8831] [Q&A] [answered]
Hello!

Is there an API method(s) for finding users with a particular set of roles?

Thank you.

**Comment by hifabienne** [comment, marked as answer]
Hi @ashep 
No not directly.
You could do a search for user grants with the role: https://zitadel.com/docs/apis/resources/mgmt/management-service-list-user-grants
And then search for all the users ids here: https://zitadel.com/docs/apis/resources/user_service_v2/user-service-list-users

**Reply by ashep** [reply]
Thank you.

**Comment by VilleFTW** [comment]
Has there not been any development/improvement in this regard? Or is the selected answer still the recommended choice? @hifabienne 

Thank you!

**Reply by rajatcing** [reply]
hey @VilleFTW  you can do find user details now with exact matching roles that you want with [ListAuthorizations](https://zitadel.com/docs/apis/resources/authorization_service_v2/zitadel-authorization-v-2-beta-authorization-service-list-authorizations). The query below searches for users with role `admin` across all orgs on your instance. I curated this curl so that you can quickly test it. You can play around with filters for various kind of search queries. I hope this helps.
```
curl --location '{{instance_url}}/zitadel.authorization.v2beta.AuthorizationService/ListAuthorizations' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'Authorization: Bearer {{access_token}}' \
--data '{
  "pagination": {
    "limit": 0,
    "asc": true
  },
  "filters": [
    {
      "roleKey": {
        "key": "admin",
        "method": "TEXT_FILTER_METHOD_EQUALS"
      }
    }
  ]
}'
```

**Reply by VilleFTW** [reply]
> hey @VilleFTW you can do find user details now with exact matching roles that you want with [ListAuthorizations](https://zitadel.com/docs/apis/resources/authorization_service_v2/zitadel-authorization-v-2-beta-authorization-service-list-authorizations). The query below searches for users with role `admin` across all orgs on your instance. I curated this curl so that you can quickly test it. You can play around with filters for various kind of search queries. I hope this helps.
> 
> ```
> curl --location '{{instance_url}}/zitadel.authorization.v2beta.AuthorizationService/ListAuthorizations' \
> --header 'Content-Type: application/json' \
> --header 'Accept: application/json' \
> --header 'Authorization: Bearer {{access_token}}' \
> --data '{
>   "pagination": {
>     "limit": 0,
>     "asc": true
>   },
>   "filters": [
>     {
>       "roleKey": {
>         "key": "admin",
>         "method": "TEXT_FILTER_METHOD_EQUALS"
>       }
>     }
>   ]
> }'
> ```

Thanks a bunch @rajatcing ! 

Piggybacking on your comment, any idea when the V2 (stable or beta) Authorization Service will be available on [Zitadel SDK for .NET](https://github.com/smartive/zitadel-net)?  I have already created a few issues there as currently there's a lack of documentation into initializing the V2 clients as part of V4 release of Zitadel. This would be highly appreciated!
